src/pretix/static/pretixpresale/js/ui/questions.js

```diff
@@ -380,26 +380,32 @@ function questions_init_profiles(el) {
             return;
         }
 
-        var selectedProfile = matchedProfiles[0];
         var $select = $(".profile-select", scope);
         var $button = $(".profile-apply", scope);
         var $help = $(".profile-desc", scope);
 
-        if (matchedProfiles.length == 1) {
+        if (matchedProfiles.length === 1) {
             $(".profile-select-control", scope).hide().parent().addClass("form-control-text");
-            $help.html(describeProfileHTML(selectedProfile)).addClass("single-profile-desc").after($button);
+            $help.html(describeProfileHTML(matchedProfiles[0])).addClass("single-profile-desc").after($button);
         }
         else {
+            var i = 0;
             for (p of matchedProfiles) {
-                $select.append("<option>" + labelForProfile(p, matchedProfiles, scope) + "</option>");
+                $select.append($("<option>").text(labelForProfile(p, matchedProfiles, scope)).attr("value", i));
```

Comment:
Good catch with the `.text()`. The value-Attribute also makes it more robust instead of using the selectedIndex.